src/common/robot.rs

```diff
@@ -121,6 +133,105 @@ impl LocalRobot {
         }
         Ok(robot)
     }
+
+    pub fn new_from_client(
+        client: &mut GrpcClient,
+        config: AppClientConfig,
+    ) -> anyhow::Result<Self> {
+        let mut robot = LocalRobot {
+            resources: ResourceMap::new(),
+        };
+
+        let conf = fetch_config(client, &config)?;
+        let components = &(conf.config.as_ref().unwrap().components);
+
+        let r = components.iter().find(|x| x.r#type == "board");
+        let b = match r {
+            Some(r) => {
+                let mut full_model = r.model.to_string();
+                if !full_model.starts_with("micro-rdk:builtin") {
+                    anyhow::bail!(
+                        "model name must be prefixed with 'micro-rdk:builtin', model name is {:?}",
+                        full_model
+                    );
+                }
+                let model = full_model.split_off(18);
```

Comment:
this would panic if there is an absence of model. Would use split or some other iterator base functions to extract the model. Or add a bound check

Reply:
if there was no model, wouldn't it fail the condition above at L.152?

Reply:
If the model is empty would the string be micro_rdk:builtin:? Or maybe it can never be the case ^^